crates/stc_ts_file_analyzer/src/analyzer/scope/mod.rs

```diff
@@ -1151,8 +1151,10 @@ impl Analyzer<'_, '_> {
             }
         }
 
-        if let Ok(ty) = self.env.get_global_type(DUMMY_SP, name.sym()) {
-            return Ok(Some(ItemRef::Owned(vec![ty].into_iter())));
+        if !self.ctx.is_type_predicate {
+            if let Ok(ty) = self.env.get_global_type(DUMMY_SP, name.sym()) {
```

Comment:
As far as I can tell this is the function that converts the type predicates to interfaces so thats why I added the if check here